### ISSUE TITLE
Missing configfiles.

### ISSUE THREAD

**Issue by Heltzen** [motoz/PellMon#73]
hi.
First time with a Raspberry. 
Tried to install pillmon, but are missing the configfiles.

I've have had the program to work but changed a configfile and then I only got a white screen when I went to the IP Address, could not fix it so tried to uninstall and reinstall. But now am i missing alle the configfiles, the folders are created but they are empty.

**Comment by motoz** [comment]
I assume you installed the .deb file and not from the sourcecode?

The debian package manage system treats configuration files differently from other files belonging to a package, mainly it tries to preserve changes you make to configfiles when upgrading or reinstalling a package. One possible explanation to what happened here is that you have erased the configuration files manually, then uninstalled the pellmon package. Dpkg then remebers that your own version of the pellmon configuration files was that all files was erased and preserves that when you reinstall it. If that is the case then you need to tell dpkg to 'completely remove pellmon', ie. remove the configuration files including the ones that have beed modified. That's as simple as:

   dpkg --purge pellmon

Then you reinstall the normal way.

A good way to keep track of your own changes in the config files is to copy the line that you want to change, comment out the original and edit the copy. If in doubt you can always look here https://github.com/motoz/PellMon/tree/master/src/conf.d to see what they looked like originally.

**Comment by Heltzen** [comment]
Thanks now it is working

I am trying to set up ramdisk for the rrd database on the pi but a fstab file is already in the folder etc

it contains 
proc					/proc	proc		deafaults				0	0
/dev/mmcblk0p6		/boot	vfat		deafaults				0	2
/dev/mmcblk0p7		/		ext4		deafaults,noatime		0	1
#a swapfile is not a swap partition, no line here
#	use dphys-swapfile [on/of] for that

when i write your line in nano i get a writing error when i try to save. 

Med venlig hilsen
	Jesper Heltzen

**Comment by motoz** [comment]
Yes, the file /etc/fstab already exists. You should add a new row as explained in the wiki. The file is owned by root so you need to run nano as the root user by using the sudo command:
`sudo nano /etc/fstab`
then you will be able to save the file.

**Comment by Heltzen** [comment]
rampdisk is mounted

I change the database to database = /mnt/ramdisk/rrd.db
And removing the hash in front of persistent_db = /usr/local/var/lib/pellmon/nvrrd.db 
There was a slight change in the path 

I restarted pellmonsrv 

Now when i start go to the Ip-asres i can se the UI but the burner says loading and all the numbers is --

Med venlig hilsen
	Jesper Heltzen

**Comment by motoz** [comment]
Did you install the ws4py dependency according to https://github.com/motoz/PellMon/wiki/How-to-install-PellMon?

**Comment by Heltzen** [comment]
yes it is installed by
sudo apt-get install python-pip
sudo pip install ws4py
Med venlig hilsen
	Jesper Heltzen

**Comment by motoz** [comment]
OK, then what plugins do you have activated? The standard system image contains references to data items that exist only in the ScotteCom plugin (for a scotte/woody pellet burner). If you use something else then you need to edit the system image to suit your needs. If you do have a scotte burner connected, are the parameters visible and working in the parameters view?

**Comment by Heltzen** [comment]
i just have the standart plugins activated. P01, P06, P08 and P09

The system works fine without using a ramdisk, i am using a scotte burner.
In parameters am i getting the message "Pellmonsrv not running?"

Med venlig hilsen
	Jesper Heltzen

**Comment by motoz** [comment]
Then you have made an error in the ramdisk configuration, which prevents pellmonsrv from running. Pellmonweb is then unable to query pellmonsrv for values for the parameters. 

You have this configuration now?

    database = /mnt/ramdisk/rrd.db
    persistent_db = /usr/local/var/lib/pellmon/nvrrd.db

What's the output of `ls -l /mnt/ramdisk` in a terminal?
What's in the event log after you restart pellmon?

**Comment by Heltzen** [comment]
Yes the test now is exactly as described 

before 
> database = /mnt/ramdisk/rrd.db
> persistent_db = /var/lib/pellmon/nvrrd.db
now
> database = /mnt/ramdisk/rrd.db
> persistent_db = /usr/local/var/lib/pellmon/nvrrd.db
same effect when i open Pellmon

The output from terminal is 
totalt 4
-rw-r--r-- 1 pellmonsrv nogroup 4096 jan 19 11:57 rrd.db

events from pellmon
2017-01-19 12:05:31,085 - INFO - copied /usr/local/var/lib/pellmon/nvrrd.db to /mnt/ramdisk/rrd.db <http://10.0.1.12:8081/#>
2017-01-19 12:05:31,067 - INFO - Created rrd database: rrdtool create /usr/local/var/lib/pellmon/nvrrd.db --step 30 DS:data1:GAUGE:120:U:U DS:data2:GAUGE:120:U:U DS:data3:GAUGE:120:U:U DS:data4:GAUGE:120:U:U DS:data5:GAUGE:120:U:U DS:data6:GAUGE:120:U:U DS:data7:GAUGE:120:U:U DS:data8:GAUGE:120:U:U DS:data9:GAUGE:120:U:U DS:data10:GAUGE:120:U:U DS:data11:GAUGE:120:U:U DS:data12:GAUGE:120:U:U DS:data13:GAUGE:120:U:U DS:data14:GAUGE:120:U:U DS:data15:GAUGE:120:U:U DS:feedertime:DERIVE:120:0:U DS:feedercapacity:GAUGE:120:U:U DS:logtick:COUNTER:120:U:U RRA:AVERAGE:0,999:1:20000 RRA:AVERAGE:0,999:10:20000 RRA:AVERAGE:0,999:100:20000 RRA:AVERAGE:0,999:1000:20000 <http://10.0.1.12:8081/#>
2017-01-19 12:05:31,024 - INFO - Activated plugins: ScotteCom, SiloLevel, Consumption, Cleaning <http://10.0.1.12:8081/#>
2017-01-19 12:05:30,073 - INFO - chip version detected as: 6.34 <http://10.0.1.12:8081/#>
2017-01-19 12:05:29,600 - INFO - serial port ok <http://10.0.1.12:8081/#>
2017-01-19 12:05:29,317 - INFO - starting pelletMonitor <http://10.0.1.12:8081/#>
2017-01-19 12:05:25,594 - INFO - deactivated Cleaning <http://10.0.1.12:8081/#>

Med venlig hilsen
	Jesper Heltzen

**Comment by motoz** [comment]
Something is not right here, your rrd.db database file in /mnt/ramdisk is way too small, only 4096 bytes. Btw, you didn't need to change the path for persistent_db, just remove the '#' sign to uncomment it. As far as I remember it should still work when placed in /usr/local/var... instead of /var but you could try changing it back. What's the output of `ls -l /usr/local/var/lib/pellmon/` ?

**Comment by Heltzen** [comment]
changed persistent_db = /var/lib/pellmon/nvrrd.db

terminal output is 
totalt 4
-rw-r--r-- 1 pellmonsrv nogroup 3072 jan 19 12:05 pellmon_settings.db

I found a mistake in /etc/fstab 

insted of
 tempfs /mnt/ramdisk tmpfs defaults,noatime,size=20M,mode=1777 0 0

I wrote
tempfs /mnt/ramdisk tmpfs defaults,noatime,size=20,mode=1777 0 0 

That solved the problem, it was the filesize there tricked med that the size was wrong. 
Is setting up on a headless Raspberry so have to write the commands in. 😄 

Thanks for the help.

terminal output is now
totalt 11268
-rw-r--r-- 1 pellmonsrv nogroup     3072 jan 19 12:05 pellmon_settings.db
-rw-r--r-- 1 pellmonsrv nogroup 11530528 jan 19 12:46 rrd.db

Med venlig hilsen
	Jesper Heltzen
